ruma-api-macros/src/util.rs

```diff
@@ -2,10 +2,62 @@
 
 use proc_macro2::{Span, TokenStream};
 use quote::quote;
-use syn::Ident;
+use std::collections::BTreeSet;
+use syn::{
+    AngleBracketedGenericArguments, GenericArgument, Ident, Lifetime,
+    ParenthesizedGenericArguments, PathArguments, Type, TypePath, TypeReference,
+};
 
 use crate::api::{metadata::Metadata, request::Request};
 
+pub fn copy_lifetime_ident(lifetimes: &mut BTreeSet<Lifetime>, ty: &Type) {
```

Comment:
Maybe `collect_lifetime_ident` instead? `copy` seems to imply copying to some other type / impl block / whatever.